From 9582d40db47da9d3a6917422f23dddc7aa8623bd Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:13 -0400
Subject: [tools] perf: Use asprintf instead of malloc plus snprintf

Message-id: <1408953210-26343-66-git-send-email-jolsa@redhat.com>
Patchwork-id: 88799
O-Subject: [PATCH RHEL7.1 BZ1133083 065/322] perf tools: Use asprintf instead of malloc plus snprintf
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit d88938ebc97df72320609b33aa0cf9a9a98accb1
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Tue Dec 3 09:23:06 2013 +0200

description
===========
The asprintf library function is equivalent to malloc plus snprintf so
use it because it is simpler.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/srcline.c b/tools/perf/util/srcline.c
index d11aefb..4c8e816 100644
--- a/tools/perf/util/srcline.c
+++ b/tools/perf/util/srcline.c
@@ -227,7 +227,6 @@ char *get_srcline(struct dso *dso, unsigned long addr)
  unsigned line = 0;
  char *srcline;
  char *dso_name = dso->long_name;
- size_t size;
 
  if (!dso->has_srcline)
   return SRCLINE_UNKNOWN;
@@ -241,13 +240,7 @@ char *get_srcline(struct dso *dso, unsigned long addr)
  if (!addr2line(dso_name, addr, &file, &line))
   goto out;
 
- /* just calculate actual length */
- size = snprintf(NULL, 0, "%s:%u", file, line) + 1;
-
- srcline = malloc(size);
- if (srcline)
-  snprintf(srcline, size, "%s:%u", file, line);
- else
+ if (asprintf(&srcline, "%s:%u", file, line) < 0)
   srcline = SRCLINE_UNKNOWN;
 
  free(file);
-- 
1.7.1